storage/replica_raftstorage.go

```diff
@@ -351,7 +351,10 @@ func (r *Replica) updateRangeInfo() error {
 	// Load the system config.
 	cfg := r.rm.Gossip().GetSystemConfig()
 	if cfg == nil {
-		return fmt.Errorf("no system config available")
+		// This could be before the system config was ever gossiped,
+		// or it expired. Let the gossip callback set the info.
+		log.Warningf("no system config available, cannot determine range MaxBytes")
```

Comment:
well, because in this case `updateRangeInfo` didn't actually do its work, so it did not complete without error. I think what we're saying is that failure in this function is sometimes acceptable. We can either model that as a particular error that is safe to ignore, or break the function into two: one which is allowed to fail, and one which is not.